pkg/integration/tests/commit/commit_with_non_matching_branch.go

```diff
@@ -0,0 +1,47 @@
+package commit
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var CommitWithNonMatchingBranchName = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Commit with defined config commitPrefixes",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig: func(testConfig *config.AppConfig) {
+		testConfig.UserConfig.Git.CommitPrefixes = map[string]config.CommitPrefixConfig{"repo": {Pattern: "^\\w+\\/(\\w+-\\w+).*", Replace: "[$1]: "}}
+	},
+	SetupRepo: func(shell *Shell) {
+		shell.NewBranch("branchnomatch")
+		shell.CreateFile("test-commit-prefix", "This is foo bar")
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		t.Views().Commits().
+			IsEmpty()
+
+		t.Views().Files().
+			IsFocused().
+			PressPrimaryAction().
+			Press(keys.Files.CommitChanges)
+
+		t.ExpectPopup().CommitMessagePanel().
+			Title(Equals("Commit summary")).
+			InitialText(Equals("")).
```

Comment:
Can't we end the test after this line? Everything that follows is not really related to what we want to test here, or is it?

Reply:
We probably could. The intent behind keeping the edit and then reopen of the popup was just to make sure that in a case where a user were to have typed something, closed  the popup and then reopened - that the initial text would remain. However I guess this is already covered in commit_wip_with_prefix, so I'll remove it